runtimes/mlflow/tests/test_metadata.py

```diff
@@ -0,0 +1,217 @@
+            ),
+            [
+                MetadataTensor(
+                    name="foo",
+                    datatype="INT32",
+                    shape=[2, 2],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                )
+            ],
+        ),
+        (
+            Schema(inputs=[TensorSpec(shape=(2, 2), type=np.dtype("int32"))]),
+            [
+                MetadataTensor(
+                    name="input-0",
+                    datatype="INT32",
+                    shape=[2, 2],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                )
+            ],
+        ),
+        (
+            Schema(
+                inputs=[
+                    ColSpec(type=DataType.string),
+                    ColSpec(type=DataType.integer),
+                ]
+            ),
+            [
+                MetadataTensor(
+                    name="input-0",
+                    datatype="BYTES",
+                    shape=[-1],
+                    tags=Tags(content_type=StringCodec.ContentType),
+                ),
+                MetadataTensor(
+                    name="input-1",
+                    datatype="INT32",
+                    shape=[-1],
+                    tags=Tags(content_type=NumpyCodec.ContentType),
+                ),
+            ],
+        ),
+    ],
+)
+def test_to_metadata_tensors(schema: Schema, expected: List[MetadataTensor]):
+    metadata_tensors = to_metadata_tensors(schema)
+
+    assert metadata_tensors == expected
+
+
+@pytest.mark.parametrize(
+    "tensor_spec, request_input",
+    [
+        (
+            ColSpec(name="foo", type=DataType.boolean),
+            RequestInput(
+                name="foo",
+                datatype="BOOL",
+                parameters=Parameters(content_type=NumpyCodec.ContentType),
+                shape=[2],
+                data=[True, False],
+            ),
+        ),
+        (
+            ColSpec(name="foo", type=DataType.string),
+            RequestInput(
+                name="foo",
+                datatype="BYTES",
+                parameters=Parameters(content_type=StringCodec.ContentType),
+                shape=[2, 11],
+                data=b"hello worldhello world",
+            ),
+        ),
+        (
+            ColSpec(name="foo", type=DataType.binary),
+            RequestInput(
+                name="foo",
+                datatype="BYTES",
+                parameters=Parameters(content_type=Base64Codec.ContentType),
+                shape=[1, 20],
+                data=b"UHl0aG9uIGlzIGZ1bg==",
+            ),
+        ),
+        (
+            ColSpec(name="foo", type=DataType.datetime),
+            RequestInput(
+                name="foo",
+                datatype="BYTES",
+                parameters=Parameters(content_type=DatetimeCodec.ContentType),
+                shape=[1, 19],
+                data=b"2021-08-24T15:01:19",
```

Comment:
should we have a case with timezone aware datetime?

Reply:
Yuuup, good point. I'll add one.